### PR TITLE
Improved the documentation.

### PR DESCRIPTION
**Pull request by pelson**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,225 @@
...
+ - python
+
+EOF
+
+git add env.spec
+git commit -m "Added the default environment."
+
+```
+
+The update process can be broken into multiple phases, any/all of which can potenitally be fully automated.
```
Comment (rhattersley) [inline README.md]: Typo in "potenitally"

**Comment by rhattersley** [comment]
Very helpful. :+1:

**Comment by pelson** [comment]
I think we can merge this when you're ready @rhattersley. I'll firm up some plans and update the implementation alongside tweaks to the docs.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,247 @@
...
+
+There is some machinery which helps us move through a next -> current -> previous workflow, but this is
+likely to change in the future. Please raise an issue if you would like more detail on this.
+
+Notes
+-----
+
+* If using ``conda-env-tracker`` on a local git repository, it will not be possible to push changes to a branch which is checked out. In these situations it is safest to put your local repo into "detached head" mode (one option ``git checkout --detach``).
+
+* Assumes a basic approach of merge and fix, rather than verify before merge. To mitigate this concern, the labels concept allows us to point to a tagand would not break if we merged something erroneously.
```
Comment (rhattersley) [inline README.md]: "tagand" -> "tag and"

Comment (pelson) [inline README.md]: Good spot.

File: README.md [hunk 3]
```diff
@@ -1,2 +1,249 @@
...
+likely to change in the future. Please raise an issue if you would like more detail on this.
+
+Notes
+-----
+
+* If using ``conda-env-tracker`` on a local git repository, it will not be possible to push changes to a branch which is checked out.
+  In these situations it is safest to put your local repo into "detached head" mode (one option ``git checkout --detach``).
+
+* Assumes a basic approach of merge and fix, rather than verify before merge. To mitigate this concern, the labels concept allows
+  us to point to a tagand would not break if we merged something erroneously.
```
Comment (rhattersley) [inline README.md]: You still have "tagand"! :confused:

Comment (pelson) [inline README.md]: Vim was doing some weird things there for me, sorry about that. In truth that whole sentence doesn't make a whole lot of sense, but I will leave that for another PR.

Comment (rhattersley) [inline README.md]: > In truth that whole sentence doesn't make a whole lot of sense

Maybe it's not the easiest to read, but it does make sense ... at least to me. :wink:

**Comment by rhattersley** [comment]
Super duper :+1:

**Pull request merged**